release-deployment.md

```diff
@@ -269,3 +273,12 @@ The process for tagging and merging is a bit different when deploying a bundle,
    Click `Publish release`.
 
    **NOTE: the release tag will be 1 commit ahead of the bundle commit**
+
+1. Merge `master` into `develop`.
```

Comment:
A good call @jansepar . I'll update the steps in this part of the doc by linking up to the sections they are referencing.